parkings/api/operator/permit.py

```diff
@@ -0,0 +1,38 @@
+from ..common_permit import PermitSerializer, PermitViewSet
+from .operator_permission import OperatorApiHasPermission
+
+
+class DomainNotRequiredMixin():
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.fields['domain'].required = False
+
+
+class DomainToCodeMixin():
+    def to_representation(self, instance):
+        representation = super().to_representation(instance)
+        representation.update(domain=instance.domain.code)
+        return representation
+
+
+class OperatorPermitSerializer(
+    DomainNotRequiredMixin,
+    DomainToCodeMixin,
+    PermitSerializer
+):
+    class Meta(PermitSerializer.Meta):
+        fields = PermitSerializer.Meta.fields + ['domain']
+
+
+class FilterUserOwnedDomainPermitMixin():
+    def get_queryset(self):
+        return (
+            super()
+            .get_queryset()
+            .filter(domain__enforcer__user=self.request.user)
```

Comment:
Operators should see all their permits. It has nothing to do with enforcers.

You should filter based on the PermitSeries.owner field:
```python
qs = super().get_queryset()
return qs.filter(series__owner=self.request.user)
```

Reply:
Since the tests are passing. This also means that there is no good test coverage for cases like this, so they should probably be updated and some new tests added too.

Reply:
E.g. here's a couple test cases for this.

Test case 1:

1. Operator A creates a PermitSeries PS1 and permits P1 and P2 into it.
2. Operator B creates a PermitSeries PS2 and permits P3 and P4 into it.
3. Operator A fetches list of all permits. It should see exactly P1 and P2.
4. Operator B fetches list of all permits. It should see exactly P3 and P4.

Test case 2:

Setup: There is an operator A, an enforcement domains X and Y, and an enforcer E which is enforcing the X.
1. Operator A creates a PermitSeries PS1 and permits P1 and P2 into it, with P1.domain = X, P2.domain = Y and they are both for registration_number ABC-123.
2. Enforcer E creates a PermitSeries PS2 and permits P3 and P4 into it. *Note*: Happens through Enforcement API, so the domains are set automatically as P3.domain = P4.domain = X. Registration number for these too is ABC-123.
3. Operator A fetches list of all permits. It should see exactly P1 and P2.
4. Enforcer E fetches list of all permits through the permit management API (/enforcement/v1/permit/). It should see exactly P3 and P4. (The operator Permits are not listed here, because this endpoint is not meant for checking the permits, but for managing the enforcers own permits.)
5. (*Later* - when the checking endpoint is updated) Enforcer E fetches the list of valid parkings for ABC-123. The result should include data from permits P1, P3 and P4, but *not* P2, since it's for another domain.

Reply:
The second last test, number 4 is not included in this PR. I think it should belong to separate commit as we will be making changes to `EnforcementPermitViewSet`.